### DISCUSSION TITLE
Design suggestion: Compile time reflection

### DISCUSSION THREAD

**Discussion by harabari-alten** [carbon-language/carbon-lang#1979] [Language design]
Compile time reflection is highly sought feature in other languages and it simplifies greatly some essential but monotonous tasks like dumping an object to string, serialisation and other stuff that can be solved by code generation. 

In my opinion this is a very important feature that will make Carbon more than just syntax sugar on top of C++.

**Comment by geoffromer** [comment]
This is definitely something we're interested in, but my sense is that it will be difficult to pursue this right now, because so much of the rest of the language is still in flux, and we don't really have a design for metaprogramming.

(I'm moving this to a discussion because it's not an issue that we can resolve with a commit in the near term)
